### PR TITLE
Prepare UCX rpm for Fedora packaging

### PR DESCRIPTION
**Pull request by amaslenn**
Rework existing rpm spec to follow Fedora's requirements.

### CODE REVIEW

**Comment by amaslenn** [comment]
@alinask @yosefe please review.

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by yosefe** [comment]
@amaslenn pls rebase on master

**Comment by shamisp** [comment]
Just for your reference https://build.opensuse.org/package/view_file/openSUSE:Factory/openucx/openucx.spec?expand=1

**Comment by alinask** [comment]
bot:retest

**Comment by amaslenn** [comment]
@shamisp thank you!

@yosefe @alinask in opensuse's approach there are multiple packages (per component: `libucp-devel`, `libucm-devel`, etc.). Should we follow this approach?
Currently I have two packages: `ucx` (shared libs, bins, perftest and `README`) and `ucx-devel` (includes headers, static libs and `pkgconfig` file).

**Comment by yosefe** [comment]
i think ucx and ucx-devel is better

**Comment by shamisp** [comment]
@yosefe - it depends what direction we want to take it. 
People that just use UCT don't have to install UCP.

**Comment by shamisp** [comment]
IMHO SuSE way is a bit cleaner from RPM standpoint, even so for a user it might be a bit confusing what to install

**Comment by miked-mellanox** [comment]
@shamisp - i thing having and managing single -devel package is simplest option, if we hear user-requests for separate -devel packages, we can consider it. for now - it is fine IMO.

**Comment by shamisp** [comment]
I have no objection.

File: ucx.spec.in [hunk 1]
```diff
@@ -1,98 +1,67 @@
...
 %description
 UCX is a communication library implementing high-performance messaging.
 
+%package devel
+Summary: Header files and libraries required to develop with ucx
+Requires: %{name}%{?_isa} = %{version}-%{release}
+
+%description devel
+UCX is a communication library implementing high-performance messaging.
+Requires either RDMA-capable device, or Cray Gemini, for inter-node
```
Comment (shamisp) [inline ucx.spec.in]: Cray Gemini or Aries

Comment (shamisp) [inline ucx.spec.in]: You may want to mention intra-node communication support with various flavors of shared memory mechanism, including sysv, cma, knem, xpmem, etc.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4044/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4048/ for details (Mellanox internal link).

File: ucx.spec.in (hunk 1 above)
Comment (amaslenn) [inline ucx.spec.in]: @shamisp thanks. Could you please suggest exact formulation for "shared memory mechanisms"? I'm not sure what will be both short enough and good for users.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4049/ for details (Mellanox internal link).

File: ucx.spec.in (hunk 1 above)
Comment (shamisp) [inline ucx.spec.in]: In addition, the library can be used for intra-node communication by leveraging the following shared memory mechanisms: posix. sysv, cma, knem, xpmem.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4051/ for details (Mellanox internal link).

**Comment by alinask** [comment]
bot:retest

**Comment by alinask** [comment]
@MattBBaker Can you please post the ORNL failure?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4076/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4075/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4081/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
@alinask Under the 'Build rpms' step.
```
Processing files: ucx-1.3.3220-1.x86_64
error: File must begin with "/": BSD
error: File must begin with "/": LICENSE
Executing(%doc): /bin/sh -e /var/tmp/rpm-tmp.nae3GY
+ umask 022
+ cd /var/lib/jenkins/jobs/UCX/workspace/build-tests
+ cd ucx-1.3.3220
+ DOCDIR=/var/lib/jenkins/jobs/UCX/workspace/build-tests/_rpm/usr/share/doc/ucx-1.3.3220
+ export DOCDIR
+ rm -rf /var/lib/jenkins/jobs/UCX/workspace/build-tests/_rpm/usr/share/doc/ucx-1.3.3220
+ /bin/mkdir -p /var/lib/jenkins/jobs/UCX/workspace/build-tests/_rpm/usr/share/doc/ucx-1.3.3220
+ cp -pr README AUTHORS NEWS /var/lib/jenkins/jobs/UCX/workspace/build-tests/_rpm/usr/share/doc/ucx-1.3.3220
+ exit 0


RPM build errors:
    File must begin with "/": BSD
    File must begin with "/": LICENSE
Build step 'Execute shell' marked build as failure
```

**Comment by alinask** [comment]
@MattBBaker thanks!

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4084/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4086/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:bgate:retest

**Comment by alinask** [comment]
@yosefe @shamisp Can you please review?
@amaslenn I think it's better to squash all the commits (after final review).

File: ucx.spec.in [hunk 2]
```diff
@@ -1,98 +1,80 @@
...
 UCX is a communication library implementing high-performance messaging.
 
+%package devel
+Summary: Header files and libraries required to develop with ucx
+Group: Development/Libraries
+Requires: %{name}%{?_isa} = %{version}-%{release}
+
+%description devel
+UCX is a communication library implementing high-performance messaging.
+Requires either RDMA-capable device, Cray Gemini or Aries, for inter-node
```
Comment (shamisp) [inline ucx.spec.in]: RDMA-capable device (InfiniBand, RoCE, etc), Cray Gemini or Aries

**Comment by shamisp** [comment]
Please squash

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4129/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4132/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
The spec file looks good. Please review the failures. I don't have any additional comments.

**Comment by alinask** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4142/ for details (Mellanox internal link).

**Pull request merged**